File: api/app/schemas/display.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class DisplayKeyword(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    text: str
    weight: float
    category: str | None = None
    display_part: str | None = Field(default=None, alias="displayPart")


class DisplaySnapshotResponse(BaseModel):
    eventSlug: str
    participantCount: int
    completedCount: int
    topMindKeywords: list[DisplayKeyword]
    topSupportKeywords: list[DisplayKeyword]
    cloudKeywords: list[DisplayKeyword]
    generatedAt: datetime

File: web/src/types/display.ts
export type DisplayKeyword = {
  text: string;
  weight: number;
  category?: string | null;
};

export type DisplaySnapshot = {
  eventSlug: string;
  participantCount: number;
  completedCount: number;
  topMindKeywords: DisplayKeyword[];
  topSupportKeywords: DisplayKeyword[];
  cloudKeywords: DisplayKeyword[];
  generatedAt: string;
};

export type DisplayConnectionStatus =
  | "connecting"
  | "connected"
  | "reconnecting"
  | "polling"
  | "disconnected";
